app/src/main/java/org/xbmc/kore/service/library/SyncMusic.java

```diff
@@ -347,12 +348,28 @@ public void onSuccess(ApiList<AudioType.DetailsSong> result) {
 
                 // Save partial results to DB
                 ContentValues songValuesBatch[] = new ContentValues[items.size()];
+                int artistsCount = 0;
                 for (int i = 0; i < items.size(); i++) {
                     AudioType.DetailsSong song = items.get(i);
                     songValuesBatch[i] = SyncUtils.contentValuesFromSong(hostId, song);
+
+                    artistsCount += song.artistid.size();
                 }
                 contentResolver.bulkInsert(MediaContract.Songs.CONTENT_URI, songValuesBatch);
 
+                // Iterate on each song, collect the artists and insert them
+                ContentValues songArtistsValuesBatch[] = new ContentValues[artistsCount];
+                int artistCount = 0;
+                for (AudioType.DetailsSong song : items) {
+                    for (int artistId : song.artistid) {
+                        songArtistsValuesBatch[artistCount] = new ContentValues();
+                        songArtistsValuesBatch[artistCount].put(MediaContract.SongArtists.HOST_ID, hostId);
+                        songArtistsValuesBatch[artistCount].put(MediaContract.SongArtists.SONGID, song.songid);
+                        songArtistsValuesBatch[artistCount].put(MediaContract.SongArtists.ARTISTID, artistId);
+                        artistCount++;
+                    }
+                }
+
```

Comment:
OK, I messed up. I originally implemented the solution on #251 and extracted the solution without the unittest into this PR. As it has been a while since I looked at this code I need more time to get this right. I'll post a comment when (I think) I fixed the code. This also puts #251 on hold.